node/overseer/src/lib.rs

```diff
@@ -680,7 +631,7 @@ where
 			.supports_parachains(supports_parachains)
 			.metrics(metrics.clone())
 			.spawner(s)
-			.build()?;
```

Comment:
I feel like in this pattern, we may want to restructure as

```
.spawner(s)
.connector(connector)
.build()
```

Reply:
`fn new` is going to die in #3874 :upside_down_face: